lib/ansible/modules/async_wrapper.py

```diff
@@ -231,14 +232,23 @@ def main():
     jobdir = os.path.expanduser(async_dir)
     job_path = os.path.join(jobdir, jid)
 
-    if not os.path.exists(jobdir):
-        try:
-            os.makedirs(jobdir)
-        except Exception:
-            print(json.dumps({
-                "failed": 1,
-                "msg": "could not create: %s" % jobdir
-            }))
+    err = None
+    try:
+        os.makedirs(jobdir)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
```

Comment:
I think this suffers from the same problem that we had discussed about 59394

This would allow someone to potentially pre-create the job dir, and using a pre-existing job dir could be considered a security vulnerability.  I think we might need to further nest the dirs, and create the parent, and use the `jid` as a subdir that we create and we ensure doesn't already exist.

Reply:
While I agree this is a good idea and I will look to see how complex it would be to add it to this PR, isn’t this a problem today? We currently do the same thing thing today expect that we don’t handle the dir being created between the check if it exists and this actually creating it. This PR simply tries to create it and just ignores if it already exists in 1 step compared to the 2 from before.

Reply:
I think there might still be a concern of a race condition, between the check, and the creation.  I know @abadger had some concerns as well.

Reply:
This PR removes the check before the creation, now the check just happens after the creation through the exception errno. So in terms of safety this doesn't add or remove anything from what's already there. If there is a security concern the concern should also exist in the current code.

Reply:
Yes.